compiler/noirc_evaluator/src/ssa/ir/instruction.rs

```diff
@@ -556,10 +556,22 @@ impl Instruction {
     /// If true the instruction will depend on `enable_side_effects` context during acir-gen.
     pub(crate) fn requires_acir_gen_predicate(&self, dfg: &DataFlowGraph) -> bool {
         match self {
-            Instruction::Binary(binary)
-                if matches!(binary.operator, BinaryOp::Div | BinaryOp::Mod) =>
-            {
-                true
+            Instruction::Binary(binary) => {
+                // Some binary math can overflow or underflow
+                match binary.operator {
```

Comment:
Let's case for just unsigned ops as we do in acir gen https://github.com/noir-lang/noir/blob/0d7642cb2071fbee148978a89a0922bfffe5be6a/compiler/noirc_evaluator/src/acir/mod.rs#L1984

Reply:
Strange, why is this check only done for unsigned types?

Reply:
Looks like we perform some overflow checks in SSA:
https://github.com/noir-lang/noir/blob/a97402aa3b3a1a7d40f61242e02d3772175846f2/compiler/noirc_evaluator/src/ssa/ssa_gen/context.rs#L344

We deferred those specific ACIR gen checks in this PR https://github.com/noir-lang/noir/pull/4832 and you can view the issue in that PR's description for more context. The TLDR for why we deferred the checks is that how we perform an overflow check can differ in ACIR vs. Brillig. Rather than deferring all the way to ACIR we should still generate these checks during SSA so the instructions can benefit from the SSA optimizations. This would be another place #6841 would benefit as we could simply code gen differing SSAs immediately.